tree/dataframe/src/RDFInterfaceUtils.cxx

```diff
@@ -157,6 +157,13 @@ static ParsedExpression ParseRDFExpression(std::string_view expr, const ColumnNa
    // when we are done, exprWithVars willl be the same as preProcessedExpr but column names will be substituted with
    // the dummy variable names in varNames
    TString exprWithVars(preProcessedExpr);
+
+   // sort the vector usedColsAndAliases by the length of its members,
+   // so in case of friends we guarantee we never substitute a column name with another column containing it
+   // ex. without sorting when passing "x" and "fr.x", the replacer would output "var0" and "fr.var0"
+   // whereas the desired behaviour is handling them as "var0" and "var1"
```

Comment:
this explains _what_ happens in the broken case but the important part for the future reader is the _why_

Reply:
Now it is:
```cpp
   // ex. without sorting when passing "x" and "fr.x", the replacer would output "var0" and "fr.var0",
   // because it has already substituted "x", hence the "x" in "fr.x" would be recognized as "var0",
   // whereas the desired behaviour is handling them as "var0" and "var1"
```